# Carácter iónico y momento dipolar

Calcular, para CO, NO, SrO y MgO, el %CI por el método de Hanny y por el méodo de Pauling. El momento dipolar utilizando la estimación de %CI de Hanny y comparar con los valores experimentales reportados.

## a) Datos

#### 1. Electronegatividad

| Elemento | $\chi$ |
|---|---|
| O | 3.44 |
| C | 2.55 |
| N | 3.04 |
| Sr | 0.95 |
| Mg | 1.31 |

#### 2. Distancia interatómica en Angstroms

| Molécula | $d$ (Å) | Fuente |
|---|---|---|
| CO | 1.1283 | [Carbon monoxide - Webbook](https://webbook.nist.gov/cgi/cbook.cgi?ID=C630080&Units=SI&Mask=1000#Diatomic) |
| NO | 1.15077 | [Nitric oxide - Webbook](https://webbook.nist.gov/cgi/cbook.cgi?ID=C10102439&Mask=1000) |
| SrO | 1.91983 | [Strontium monoxide - Webbook](https://webbook.nist.gov/cgi/cbook.cgi?ID=C1314110&Units=SI&Mask=1000#Dia9) |
| MgO | 1.749 | [Magnesium oxide - Webbook](https://webbook.nist.gov/cgi/cbook.cgi?ID=C1309484&Units=SI&Mask=1000#ref-2) |


#### 3. Momentos dipolares (en Debye) reportados

| Molécula | $\mu$ (D) | Fuente |
|---|---|---|
| CO | 0.122 | [Carbon monoxide - Webbook](https://webbook.nist.gov/cgi/cbook.cgi?ID=C630080&Mask=1000#Diatomic), [Carbon monoxide - Wikipedia](https://en.wikipedia.org/wiki/Carbon_monoxide) |
| NO | 0.0 | [The Dipole Moment of Nitric Oxide, Smyth and McAlpine, 2004](https://aip.scitation.org/doi/10.1063/1.1749220) |
| SrO | 8.9 | [Electronic Structure of SrO, Kaufman, Wharton, and Klemperer, 2004](https://aip.scitation.org/doi/10.1063/1.1696875) |
| MgO | 6.2 | [Electric dipole moments of the MgO B1Σ+ and X1Σ+ states, Büsener, Heinrich and Hese, 1987](https://www.sciencedirect.com/science/article/abs/pii/0301010487850292) |

In [1]:
import numpy as np
from tabulate import tabulate

In [2]:
# Electronegatividad
X = {"CO": {"Xa": 2.55, "Xb": 3.44}, "NO": {"Xa": 3.04, "Xb": 3.44},
     "SrO": {"Xa": 0.95, "Xb": 3.44}, "MgO": {"Xa": 1.31, "Xb": 3.44}}

# Distancia interatomica en angstroms
d = {"CO": 1.1283, "NO": 1.15077, "SrO": 1.91983, "MgO": 1.749}

# Momento dipolar en D
mu_exp = {"CO": 0.122, "NO": 0.0, "SrO": 8.9, "MgO": 6.2}

In [3]:
# Carga del electron
e = 4.8e-10

## b) Caracter iónico

El %CI puede ser estimado por medio de la ecuación propuesta por Hanny:

$$\text{%CI}_{h} = 16 \left| \chi_a - \chi_b \right| + 3.5 \left| \chi_a - \chi_b \right|^2$$

O por medio de la ecuación propuesta por Pauling:

$$\displaystyle \text{%CI}_{p} = 100 \left[ 1 - e^{-\frac{1}{4} \left( \chi_a - \chi_b \right)^2} \right]$$

In [4]:
# %CI de Hanny
def pCI_h(Xa, Xb):
    diff = np.abs(Xa-Xb)
    return 16*diff + 3.5*diff**2

In [5]:
pCI_h_ = {molec:pCI_h(**X[molec]) for molec in X}
pCI_h_

{'CO': 17.01235,
 'NO': 6.959999999999998,
 'SrO': 61.540350000000004,
 'MgO': 49.959149999999994}

In [6]:
# %CI de Pauling
def pCI_l(Xa, Xb):
    diff = np.abs(Xa-Xb)
    return 100*(1-np.exp(-0.25*diff**2))

In [7]:
pCI_l_ = {molec:pCI_l(**X[molec]) for molec in X}
pCI_l_

{'CO': 17.965065585213402,
 'NO': 3.9210560847676823,
 'SrO': 78.77573323062757,
 'MgO': 67.83286799695526}

Así, para cda compuesto tenemos:

| Molécula | %CI de Hanny | %CI de Pauling |
|---|---|---|
| CO | 17.0% | 18.0% |
| NO | 7.0% | 3.9% |
| SrO | 61.5% | 78.8% |
| MgO | 50.0% | 67.8% |

## c) Momento dipolar.

A continuación se presenta el cálculo del momento dipolar utilizando la estimación de %CI de Hanney, y finalmente se compara con los valores experimentales reportados.


In [11]:
# Momento dipolar en D
def mu(pCI, d):
    # delta_AB * d_AB
    delta = (pCI/100.0) * e
    return delta * (d * 1e-8) / (1e-18)

In [12]:
mu_ = {molec:mu(pCI_h_[molec], d[molec]) for molec in d}
mu_ 

{'CO': 0.92136165624,
 'NO': 0.38444924159999994,
 'SrO': 5.671056486744,
 'MgO': 4.194170560800001}

#### Diferencia porcentual $\mu$ vs $\mu_{exp}$

In [13]:
def p_diff(mu, mu_exp):
    if mu_exp == 0:
        return 'NA'
    return "{:0.0f}%".format(100*(mu-mu_exp)/mu_exp)

In [14]:
p_diff_ = {molec:p_diff(mu_[molec], mu_exp[molec]) for molec in mu_}
p_diff_

{'CO': '655%', 'NO': 'NA', 'SrO': '-36%', 'MgO': '-32%'}

| Molécula | $d$ (Å) | $\mu$ | $\mu_{exp}$ | $\% \Delta$ |
|---|---|---|---|---|
| CO | 1.1283 | 0.92 | 0.122 | 655% |
| NO | 1.15077 | 0.38 | 0.0 | NA |
| SrO | 1.91983 | 5.67 | 8.9 | -36% |
| MgO | 1.749 | 4.19 | 6.2 | -32% |